sdk/core/AzureCore/Source/Util/JSONPatchUtil.swift

```diff
@@ -0,0 +1,117 @@
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import Foundation
+
+enum JSONPatchOpType: String, Codable {
+    case add
+    case remove
+    case replace
+    case move
+    case copy
+    case test
+}
+
+/// Serves as a sentinel value for JSON Patch operations to nullify a property
+public struct JSONPatch {
+    public static let null = "__NULL__"
+}
+
+struct JSONPatchOperation: Encodable {
+    let operation: JSONPatchOpType
+    let from: String?
+    let path: String
+    let value: String?
+
+    // MARK: Codable
+
+    public enum CodingKeys: String, CodingKey {
+        case operation, from, path, value
+    }
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(operation, forKey: .operation)
+        try container.encode(from, forKey: .from)
+        try container.encode(path, forKey: .path)
+        try container.encode(value, forKey: .value)
+    }
+}
+
+/// Helper class for creating PatchObjects
+public final class JSONPatchObject: Encodable {
+    fileprivate var operations: [JSONPatchOperation]
+
+    public init() {
+        self.operations = [JSONPatchOperation]()
+    }
+
+//    /// Inserts a new value at an array index or adds a new property.
+//    public func add<T: Codable>(atPath path: String, withValue value: T) {
+//        operations.append(PatchOperation(operation: .add, from: nil, path: path, value: value))
+//    }
+//
+//    /// Remove the property or entry at an array index. Property must exist.
+//    public func remove(atPath path: String) {
+//        operations.append(PatchOperation(operation: .remove, from: nil, path: path, data: nil))
+//    }
```

Comment:
These are comment out for now since the only operation being used right now is `replace`.